### PR TITLE
review/jm-initial

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
This PR is intended to serve as an initial review of the Workday dbt package. This PR is not intended to be merged into `main`. Please see the review comments below and address the notes within the `main` branch. Once the comment has been addressed, please comment as such and do not resolve the comment. I will resolve the comments upon final review.

### CODE REVIEW

File: models/staging/stg_workday.yml [hunk 1]
```diff
@@ -22,8 +22,8 @@ models:
       - name: critical_job
         description: '{{ doc("critical_job") }}'
 
-      - name: job_profile_description
-        description: '{{ doc("job_profile_description") }}'
+      - name: job_profile_description
```
Comment (fivetran-joemarkiewicz) [inline models/staging/stg_workday.yml:25]: There are a few fields that were renamed in the staging models and need to be renamed in the yml. Please see the highlighted fields within the stg_workday.yml for all the ones I noticed needed to be updated.

Additionally, please note that any other fields that are renamed in the staging models (one of my previous comments) will also need to see the docs updated to reflect the new name.

File: models/staging/stg_workday__job_family_group.sql [hunk 2]
```diff
@@ -25,7 +25,7 @@ final as (
     
     select 
         source_relation, 
-        _fivetran_deleted,
+        _fivetran_deleted,
```
Comment (fivetran-joemarkiewicz) [inline models/staging/stg_workday__job_family_group.sql:28]: A large number of these staging models include the `_fivetran_deleted` field. Usually we will apply a filter to remove the deleted records in the staging models. Do you feel we should also apply that same update for Workday staging models?

File: models/staging/stg_workday__personal_information.sql [hunk 3]
```diff
@@ -36,7 +36,7 @@ final as (
         date_of_birth,
         date_of_death,
         gender,
-        hispanic_or_latino,
+        hispanic_or_latino,
```
Comment (fivetran-joemarkiewicz) [inline models/staging/stg_workday__personal_information.sql:39]: This is a boolean field and should be renamed to be `is_hispanic_or_latino`. Can you go through the fields and be sure to rename any of the boolean fields to have this same naming convention for consistency.

File: models/workday.yml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 version: 2
 
 models:
-  - name: workday__employee_overview
+  - name: workday__employee_overview
```
Comment (fivetran-joemarkiewicz) [inline models/workday.yml:4]: None of the end models have tests applied to them. Can you be sure to update the end models to include the appropriate tests.

File: models/workday.yml [hunk 5]
```diff
@@ -91,17 +91,17 @@ models:
       - name: email_address
         description: '{{ doc("email_address") }}' 
 
-      - name: employee_type
-        description: '{{ doc("employee_type") }}' 
+      - name: employee_type
```
Comment (fivetran-joemarkiewicz) [inline models/workday.yml:94]: There are a few fields that were renamed or missed in the documentation. Please review the docs and the highlighted fields within the workday.yml to see which fields need to be updated.

File: models/workday__employee_overview.sql [hunk 6]
```diff
@@ -19,7 +19,8 @@ int_worker_position_enriched as (
 
 worker_employee_enhanced as (
 
-    select *,
+    select 
+        *,
```
Comment (fivetran-joemarkiewicz) [inline models/workday__employee_overview.sql:23]: Same formatting request.

File: models/workday__employee_overview.sql [hunk 7]
```diff
@@ -31,8 +32,8 @@ worker_employee_enhanced as (
         case when days_of_employment >= 365*20 and is_user_active then true else false end as is_current_employee_twenty_years,
         case when days_of_employment >= 365*30 and is_user_active then true else false end as is_current_employee_thirty_years
     from int_worker_base
-    left join int_worker_personal_details using(worker_id)
-    left join int_worker_position_enriched using(worker_id)
+    left join int_worker_personal_details using(worker_id) 
+    left join int_worker_position_enriched using(worker_id)
```
Comment (fivetran-joemarkiewicz) [inline models/workday__employee_overview.sql:36]: Same comment about using a left join as opposed to using.

File: models/workday__organization_overview.sql [hunk 8]
```diff
@@ -1,4 +1,4 @@
-with organization_data as (
+with organization_data as (
```
Comment (fivetran-joemarkiewicz) [inline models/workday__organization_overview.sql:1]: This end model is defined as:

> Each record represents an organization, enhanced with role details.

However, it seems the grain is organization+role as opposed to just organization. Should we consider renaming this end model or updating the documentation to clearly detail the grain. What are your thoughts?

File: models/workday__position_overview.sql [hunk 9]
```diff
@@ -34,8 +34,9 @@ position_data_enhanced as (
         position_data.compensation_grade_code,
         position_data.compensation_grade_profile_code
     from position_data
-    left join position_job_profile_data on position_job_profile_data.position_id = position_data.position_id
-    where not position_data._fivetran_deleted
+    left join position_job_profile_data 
+        on position_job_profile_data.position_id = position_data.position_id
```
Comment (fivetran-joemarkiewicz) [inline models/workday__position_overview.sql:38]: Request to change the format of the left join

File: models/workday__position_overview.sql [hunk 10]
```diff
@@ -34,8 +34,9 @@ position_data_enhanced as (
...
+    where not position_data._fivetran_deleted
```
Comment (fivetran-joemarkiewicz) [inline models/workday__position_overview.sql:39]: Similar to my previous comment about filtering out the deleted records at the staging layer. Should we just do these filters there as opposed to applying them at the end models?

**Review by fivetran-joemarkiewicz — COMMENTED** [review]
@fivetran-avinash the workday model is coming together great! I have a few comments and suggestions I would like you to address below. Please apply the updates to the `main` branch and then mark them as applied or provide your comments to the comments in this PR. I can then mark the comments as resolved when doing my final review.

Great job and I am looking forward to this model going live soon!

File: models/workday__organization_overview.sql (hunk 8 above)
Comment (fivetran-joemarkiewicz) [reply models/workday__organization_overview.sql:1]: Let's consider renaming this to be `workday__role_overview` as this is more representative of the grain of the data.

File: .github/workflows/auto-release.yml [hunk 11]
```diff
@@ -9,5 +9,5 @@ on:
 jobs:
   call-workflow-passing-data:
     if: github.event.pull_request.merged
-    uses: fivetran/dbt_package_automations/.github/workflows/auto-release.yml@feature/auto-releaser
+    uses: fivetran/dbt_package_automations/.github/workflows/auto-release.yml@main
```
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply .github/workflows/auto-release.yml:12]: Applied.

File: CHANGELOG.md [hunk 12]
```diff
@@ -8,9 +8,9 @@ This package models Workday HCM data from [Fivetran's connector](https://fivetra
 
 The main focus of the package is to transform the core object tables into analytics-ready models:
 <!--section="workday_model"-->
-  - Materializes [Salesforce Marketing Cloud staging tables](https://fivetran.github.io/dbt_workday/#!/overview/workday/models/?g_v=1) which leverage data in the format described by [this ERD](https://fivetran.com/docs/applications/workday-hcm/#schemainformation). The staging tables clean, test, and prepare your Workday HCM data from [Fivetran's connector](https://fivetran.com/docs/applications/workday-hcm) for analysis by doing the following:
+  - Materializes [Salesforce Marketing Cloud staging tables](https://fivetran.github.io/dbt_workday/#!/overview/workday/models/?g_v=1) which leverage data in the format described by [this ERD](https://fivetran.com/docs/applications/workday-hcm/#schemainformation). The staging tables clean, test, and prepare your Workday HCM data from [Fivetran's connector](https://fivetran.com/docs/applications/workday-hcm) for analysis by doing the following:
```
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply CHANGELOG.md:11]: Fixed.

File: README.md [hunk 13]
```diff
@@ -34,18 +34,18 @@ The following table provides a detailed list of all models materialized within t
...
+| [workday__position_overview](https://fivetran.github.io/dbt_workday/#!/model/model.workday.workday__position_overview) | Each record represents a position with enriched data on positions. | 
 <!--section-end-->
 
 # 🎯 How do I use the dbt package?
 
 ## Step 1: Prerequisites
 To use this dbt package, you must have the following:
 
-- At least one Fivetran Workday connector syncing data into your destination.
+- At least one Fivetran Workday HCM connector syncing data into your destination.
```
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply README.md:48]: Updated. Might have added HCM after every mention of Workday, let me know if that's overkill.

File: dbt_project.yml [hunk 14]
```diff
@@ -13,7 +13,6 @@ models:
       +materialized: view
       +schema: stg_workday
 
-
```
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply dbt_project.yml:16]: Removed.

File: integration_tests/requirements.txt [hunk 15]
```diff
@@ -4,4 +4,4 @@ dbt-redshift>=1.3.0,<2.0.0
 dbt-postgres>=1.3.0,<2.0.0
 dbt-spark>=1.3.0,<2.0.0
 dbt-spark[PyHive]>=1.3.0,<2.0.0
-dbt-databricks>=1.3.0,<2.0.0
\ No newline at end of file
+dbt-databricks>=1.6.0,<2.0.0
```
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply integration_tests/requirements.txt:7]: Bumped.

File: models/intermediate/int_workday__personal_details.sql [hunk 16]
```diff
@@ -54,10 +54,10 @@ worker_personal_details as (
...
         worker_military.military_status
     from worker_personal_info_data
-    left join worker_name using(worker_id)
-    left join worker_email using(worker_id)
-    left join worker_ethnicity using(worker_id)
-    left join worker_military using(worker_id)
+    left join worker_name using(worker_id) 
+    left join worker_email using(worker_id) 
+    left join worker_ethnicity using(worker_id) 
+    left join worker_military using(worker_id)
```
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply models/intermediate/int_workday__personal_details.sql:60]: Updated.

File: models/intermediate/int_workday__worker_details.sql [hunk 17]
```diff
@@ -1,6 +1,7 @@
 with worker_data as (
 
-    select *,
+    select 
+        *,
```
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply models/intermediate/int_workday__worker_details.sql:4]: Formatted in the relevant models

File: models/intermediate/int_workday__worker_details.sql [hunk 18]
```diff
@@ -12,12 +13,14 @@ worker_details as (
...
-            and (termination_date is null or termination_date > current_date)
+        case when active 
+            then true 
+            else false 
+        end as is_user_active,
+        case when hire_date <= current_date and (termination_date is null or termination_date > current_date)
             then true 
             else false 
-            end as is_employed,
+        end as is_employed,
```
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply models/intermediate/int_workday__worker_details.sql:23]: Formats applied

File: models/intermediate/int_workday__worker_position_enriched.sql [hunk 19]
```diff
@@ -12,18 +12,18 @@ worker_position_data_enhanced as (
...
         job_profile_id,
-        case when end_date is null
-            then {{ dbt.datediff('current_date', 'start_date', 'day') }}
-            else {{ dbt.datediff('end_date', 'start_date', 'day') }}
-            end as days_at_position,
-        row_number() over (partition by worker_id order by end_date desc) as rn
+        case when end_date is null 
+            then {{ dbt.datediff('current_date', 'start_date', 'day') }} 
+            else {{ dbt.datediff('end_date', 'start_date', 'day') }} 
+            end as days_at_position,
```
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply models/intermediate/int_workday__worker_position_enriched.sql:25]: yes, oversight from porting over the BigQuery function from the drafts

File: models/intermediate/int_workday__worker_position_enriched.sql [hunk 20]
```diff
@@ -12,18 +12,18 @@ worker_position_data_enhanced as (
...
-        case when end_date is null
-            then {{ dbt.datediff('current_date', 'start_date', 'day') }}
-            else {{ dbt.datediff('end_date', 'start_date', 'day') }}
-            end as days_at_position,
-        row_number() over (partition by worker_id order by end_date desc) as rn
+        case when end_date is null 
+            then {{ dbt.datediff('current_date', 'start_date', 'day') }} 
+            else {{ dbt.datediff('end_date', 'start_date', 'day') }} 
+            end as days_at_position, 
+        row_number() over (partition by worker_id order by end_date desc) as rn
```
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply models/intermediate/int_workday__worker_position_enriched.sql:26]: Renamed.

File: models/intermediate/int_workday__worker_position_enriched.sql [hunk 21]
```diff
@@ -52,19 +52,19 @@ worker_position_enriched as (
...
-        most_recent_position.position_effective_date as most_recent_position_effective_date,
+        most_recent_position.position_start_date as most_recent_position_start_date, 
+        most_recent_position.position_end_date as most_recent_position_end_date, 
+        most_recent_position.position_effective_date as most_recent_position_effective_date, 
         worker_position_measures.worker_positions,
         worker_position_measures.worker_levels, 
         worker_position_measures.position_days
     from most_recent_position
-    left join worker_position_measures using(worker_id)
+    left join worker_position_measures using(worker_id)
```
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply models/intermediate/int_workday__worker_position_enriched.sql:67]: Fixed

File: models/workday__employee_overview.sql (hunk 6 above)
Comment (fivetran-avinash) [reply models/workday__employee_overview.sql:23]: Reformatted

File: models/workday__employee_overview.sql (hunk 7 above)
Comment (fivetran-avinash) [reply models/workday__employee_overview.sql:36]: Updated.

File: models/workday__position_overview.sql (hunk 9 above)
Comment (fivetran-avinash) [reply models/workday__position_overview.sql:38]: Sorry, what do you mean by change the format?

Comment (fivetran-joemarkiewicz) [reply models/workday__position_overview.sql:38]: On the `main` branch the left join is all on a single line

https://github.com/fivetran/dbt_workday/blob/513f6f55ca04b479890b35a35684658740784656/models/workday__position_overview.sql#L37

This suggestion is to update the format to my change above where the left join is broken out into two lines
```sql
    left join position_job_profile_data 
        on position_job_profile_data.position_id = position_data.position_id
```

Comment (fivetran-avinash) [reply models/workday__position_overview.sql:38]: Ah. Updated.

File: models/workday__position_overview.sql (hunk 10 above)
Comment (fivetran-avinash) [reply models/workday__position_overview.sql:39]: good call updated

File: models/staging/stg_workday__job_family_group.sql (hunk 2 above)
Comment (fivetran-avinash) [reply models/staging/stg_workday__job_family_group.sql:28]: Agreed, updated

File: models/staging/stg_workday__personal_information.sql (hunk 3 above)
Comment (fivetran-avinash) [reply models/staging/stg_workday__personal_information.sql:39]: Updated

File: models/staging/stg_workday.yml [hunk 22]
```diff
@@ -1,9 +1,9 @@
 version: 2
 
 models:
-  - name: stg_workday__job_profile
+  - name: stg_workday__job_profile 
     description: '{{ doc("job_profile") }}'
-    columns:
+    columns:
```
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply models/staging/stg_workday.yml:6]: Added

File: models/intermediate/int_workday__personal_details.sql [hunk 23]
```diff
@@ -1,7 +1,7 @@
 with worker_personal_info_data as(
 
     select 
-        personal_information_id as worker_id, 
+        personal_information_id as worker_id,
```
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply models/intermediate/int_workday__personal_details.sql:4]: Should be updated!

File: models/staging/stg_workday__personal_information.sql (hunk 3 above)
Comment (fivetran-avinash) [reply models/staging/stg_workday__personal_information.sql:39]: @fivetran-joemarkiewicz I wonder if we can automate this to automatically populate (if macro field is boolean, append with the renaming in the staging model)

File: models/workday__organization_overview.sql (hunk 8 above)
Comment (fivetran-avinash) [reply models/workday__organization_overview.sql:1]: Renamed.

File: models/staging/stg_workday.yml [hunk 24]
```diff
@@ -1,9 +1,9 @@
 version: 2
 
 models:
-  - name: stg_workday__job_profile
+  - name: stg_workday__job_profile
```
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply models/staging/stg_workday.yml:4]: Added.

File: models/workday.yml (hunk 4 above)
Comment (fivetran-avinash) [reply models/workday.yml:4]: Updated.

File: models/workday__organization_overview.sql (hunk 8 above)
Comment (fivetran-avinash) [reply models/workday__organization_overview.sql:1]: I decided to keep this as `workday__organization_overview` (no one is really going to look at role too closely in comparison to organization) and improved the documentation. 

I also realized doing just role and organization as the grain for the `organziation_overview` provides minimal insight as an end model. I did want to find a way to tie it back to the `employee_overview` somehow so customers can join back to those tables if they want to do their own transformations. So I also added in `worker_id` and `position_id`  to the grain, by adding a new staging model which ties worker and position back to organization (I noticed that in the new history tables there was a `worker_position_organization_history`  table I should have brought into begin with).

File: models/staging/stg_workday.yml (hunk 1 above)
Comment (fivetran-avinash) [reply models/staging/stg_workday.yml:25]: Renamed.

File: README.md [hunk 25]
```diff
@@ -34,18 +34,18 @@ The following table provides a detailed list of all models materialized within t
...
-| [workday__position_overview](https://fivetran.github.io/dbt_workday/#!/model/model.workday.workday__position_overview) | Each record represents a position with enriched data on positions. |
+| [workday__employee_overview](https://fivetran.github.io/dbt_workday/#!/model/model.workday.workday__employee_overview)  | Each record represents an employee with enriched personal information and the positions they hold.  | 
+| [workday__job_overview](https://fivetran.github.io/dbt_workday/#!/model/model.workday.workday__job_overview)  | Each record represents a job with enriched details on job profiles and job families.  | 
+| [workday__organization_overview](https://fivetran.github.io/dbt_workday/#!/model/model.workday.workday__organization_overview) |  Each record represents an organization with roles and workers.  | 
+| [workday__position_overview](https://fivetran.github.io/dbt_workday/#!/model/model.workday.workday__position_overview) | Each record represents a position with enriched data on positions. |
```
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply README.md:40]: Updated these with more details. Also added this to the CHANGELOG.

File: models/workday.yml (hunk 5 above)
Comment (fivetran-avinash) [reply models/workday.yml:94]: Should be all accounted for now.

**Comment by fivetran-avinash** [comment]
@fivetran-joemarkiewicz All PR comments addressed!

One significant change: I did a revamp of the organization model to bring in more useful data as well as make it easier to join on worker/position data for better analysis. This led to a larger grain though. Not sure if that's best practice. I'll leave more details in the Height ticket.

**Review by fivetran-avinash — COMMENTED** [review]
@fivetran-joemarkiewicz All PR comments addressed!

One significant change: I did a revamp of the organization model to bring in more useful data as well as make it easier to join on worker/position data for better analysis. This led to a larger grain though. Not sure if that's best practice. I'll leave more details in the Height ticket.

File: models/staging/stg_workday__personal_information.sql (hunk 3 above)
Comment (fivetran-joemarkiewicz) [reply models/staging/stg_workday__personal_information.sql:39]: Yes! This is something that should be easily automated for future packages. Will add as a feature to the automation repo

**Review by fivetran-joemarkiewicz — COMMENTED** [review]
Thanks @fivetran-avinash for going through all of my review notes and applying the necessary changes! This package is looking great and should be ready to go with just a few minor last minute edits. I will close out this PR, but if you can review and merge in PR #2 to address the last minute changes that would be great! 

Let me know if you have any questions. We can then discuss release plans.

**Pull request closed without merging**